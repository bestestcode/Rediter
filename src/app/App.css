html, body {
  background-color: #fafafa;
  color: #fff; }

body {
  height: 100vh;
  display: flex;
  flex-direction: column; }

.rf-new-post-notification {
  display: flex;
  justify-content: center;
  margin: 15px 0px; }

.rf-vertical-center {
  display: flex;
  align-items: center; }

.rf-timestamp {
  margin-left: 24px; }

a {
  text-decoration: none;
  transition: all .6s ease 0s; }
  a:hover {
    opacity: .6;
    color: #b1afaf;
    cursor: pointer; }

.card.horizontal {
  display: flex;
  flex-grow: 1;
  flex-wrap: wrap; }

.card .card-content {
  display: flex;
  flex-direction: column;
  justify-content: center; }

.card .card-image {
  width: 160px;
  padding: 20px; }

.card .card-expand {
  padding-left: 184px; }

.card .card-meta {
  color: #000;
  font-weight: 300; }

.card .card-author {
  color: #ef8361; }

.card .card-title-text a {
  color: #000;
  font-size: 16px;
  font-weight: 400; }

.card .card-expand {
  width: 100%;
  height: auto; }

.card .chip-list {
  display: flex; }

.card .chip {
  margin: 10px 0px 10px 12px; }

.card .action {
  margin-right: 12px; }
  .card .action a {
    color: #ff8a65; }

.card i.right {
  margin-left: 4px; }

.pagination {
  display: flex;
  color: #000; }
  .pagination .previous {
    align-content: flex-start; }
  .pagination .next {
    align-content: flex-end; }
  .pagination .middle {
    margin: 0 auto; }

.progress-container h4 {
  color: #9ea1a8;
  font-weight: 800; }

.img-responsive {
  max-width: 100%;
  height: auto;
  display: block; }

.iframe-responsive iframe {
  max-width: 100%; }

.input-field {
  color: #9e9e9e; }

.text-right {
  text-align: right; }

.container .row {
  padding-top: 20px; }
  .container .row a {
    display: inline-flex;
    padding: 0 20px; }
    .container .row a i {
      margin-right: 6px; }

table {
  color: black; }
  table a {
    padding: 4px; }
    table a.btn-icon {
      color: #ff8a65; }
