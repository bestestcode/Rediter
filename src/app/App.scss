@import '../_style/feed.scss';
@import '../_style/subreddits.scss';
