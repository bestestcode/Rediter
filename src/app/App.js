import React from 'react';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import { IndexLinkContainer, LinkContainer } from 'react-router-bootstrap';
import { Route, Router, Switch } from 'react-router';
import { Provider } from 'react-redux';
import ReduxToastr from 'react-redux-toastr';

import { history } from '../_common/helpers';
import store from '../store/store';
import Feed from '../components/feed/Feed';
import { SubredditList, SubredditAdd } from '../components/subreddits';
import SubsLoader from '../components/SubsLoader';

require('./App.css');

const App = () => (
  <Provider store={store}>
    <Router history={history}>
      <div>
        <Navbar>
          <Nav>
            <IndexLinkContainer to="/">
              <NavItem>My Feed</NavItem>
            </IndexLinkContainer>
            <LinkContainer to="/subreddits">
              <NavItem>Subreddits</NavItem>
            </LinkContainer>
          </Nav>
        </Navbar>

        <div className="container">
          <Switch>
            <Route path="/subreddits/new" component={SubredditAdd} />
            <Route path="/subreddits" component={SubredditList} />
            <SubsLoader component={Feed} />
          </Switch>
        </div>

        <ReduxToastr
          timeOut={4000}
          newestOnTop={false}
          preventDuplicates
          position="top-right"
          transitionIn="fadeIn"
          transitionOut="fadeOut"
          progressBar
        />
      </div>
    </Router>
  </Provider>
);

export default App;
