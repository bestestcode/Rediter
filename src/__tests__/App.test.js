import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';

import App from '../app/App';

describe('Root testing', () => {
  const wrapper = shallow(<App />);
  it('+++ App Dumb Component Rendering', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('+++ App Snapshot Testing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
