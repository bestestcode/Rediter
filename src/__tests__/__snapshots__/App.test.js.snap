// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root testing +++ App Snapshot Testing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div>
      <Uncontrolled(Navbar)
        bsStyle="default"
      >
        <Nav
          bsClass="nav"
          justified={false}
          pullLeft={false}
          pullRight={false}
          stacked={false}
        >
          <IndexLinkContainer
            to="/"
          >
            <NavItem
              active={false}
              disabled={false}
            >
              My Feed
            </NavItem>
          </IndexLinkContainer>
          <LinkContainer
            activeClassName="active"
            exact={false}
            replace={false}
            strict={false}
            to="/subreddits"
          >
            <NavItem
              active={false}
              disabled={false}
            >
              Subreddits
            </NavItem>
          </LinkContainer>
        </Nav>
      </Uncontrolled(Navbar)>
      <div
        className="container"
      >
        <Switch>
          <Route
            component={[Function]}
            path="/subreddits/new"
          />
          <Route
            component={[Function]}
            path="/subreddits"
          />
          <Connect(SubsLoader)
            component={[Function]}
          />
        </Switch>
      </div>
      <Connect(ReduxToastr)
        newestOnTop={false}
        position="top-right"
        preventDuplicates={true}
        progressBar={true}
        timeOut={4000}
        transitionIn="fadeIn"
        transitionOut="fadeOut"
      />
    </div>
  </Router>
</Provider>
`;
